compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -585,6 +586,24 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
+        let bound_names = trait_bounds
+            .iter()
+            .flat_map(|poly_trait_ref| {
+                poly_trait_ref.trait_ref.path.segments.iter().flat_map(|x| {
+                    x.args.iter().flat_map(|args| {
+                        args.bindings.iter().map(|binding| {
+                            let name = binding.ident.name;
+                            let trait_def = poly_trait_ref.trait_ref.path.res.opt_def_id();
+                            let assoc_item = trait_def.and_then(|did| {
+                                tcx.associated_items(did).filter_by_name_unhygienic(name).next()
```

Comment:
We shouldn't be filtering unhygenically. You have an identifier in `binding.ident`.